project/source_code/utils/src/can_io_thread.cpp

```diff
@@ -19,49 +19,37 @@ CanIOThread::CanIOThread(scpp::SocketCan *socket, std::promise<void> *promise, u
 * @param receive_message_id id of the messages to read
 * @param transmit_message_id id of the messages to transmit. 0 if nothing to transmit
 */
-void CanIOThread::Run(std::promise<void> *promise, uint8_t receive_message_id, uint8_t transmit_message_id) { //TODO break, list of IDs to listed to. Do not need the promise {
+void CanIOThread::Run(std::future<void> *future, uint8_t *receive_message_id, size_t receive_message_id_size)
+{ //TODO break, list of IDs to listed to. Do not need the promise {
     size_t i = 0;
-    while (true)
+    std::future_status future_status;
+    do
     {
+        //READ FROM SOCKET
         scpp::CanFrame fr;
         auto result = this->socket->read(fr);
         if (result == scpp::STATUS_OK)
         {
-            i = 0;
-            if (fr.id == 1) { //always stop thread if receiving "end simulation" command from user.
-                UserInput *input = reinterpret_cast<UserInput*>(fr.data);
-                if(input->end_simulation)
-                {
-                    promise->set_value();
+            for (size_t i = 0; i < receive_message_id_size; i++) {
+                if (receive_message_id[i] == fr.id) {
+                    CanBuffer::GetInstance().AddRx(&fr.id, fr.data, &fr.len);
```

Comment:
Looping the size of the received CAN frame but filling in the buffer dependent on the ID?